addrs/provider.go

```diff
@@ -126,6 +126,20 @@ func (pt Provider) LessThan(other Provider) bool {
 	}
 }
 
+// Equals returns true if the receiver and other provider have the same attributes.
+func (pt Provider) Equals(other Provider) bool {
+	if pt.Namespace != other.Namespace {
+		return false
+	}
+	if pt.Hostname != other.Hostname {
+		return false
+	}
+	if pt.Type != other.Type {
+		return false
+	}
+	return true
+}
```

Comment:
I think the above is exhaustively checking all of the fields for equality. If so, note that Go allows us to do that with the `==` operator because we're working with struct values here, rather than pointers:

```
    return pt == other
```

This is the same property that allows us to use this type as a map key, so we're already relying on it a lot.

I do like the idea of having the method to help make that explicit/consistent though, even if it does just become a shorthand for the equality check.

Reply:
👍 well i feel silly, no longer sure why I felt a need to write this whole thing 😂